### DISCUSSION TITLE
Creating custom embeddings using openAI?

### DISCUSSION THREAD

**Discussion by akarshghale** [MaartenGr/BERTopic#1214] [Q&A] [answered]
Hi,

I am looking to create embeddings using openAI. Could you please share when creating the embeddings object what should be the properties of the object?

I mean basically how should the embeddings be stored for passing into Bertopic for further processing?

**Comment by MaartenGr** [comment]
If you want to use custom embeddings, so embeddings generated outside of BERTopic, you can follow along with the [guide in the documentation](https://maartengr.github.io/BERTopic/getting_started/embeddings/embeddings.html#custom-embeddings).

**Reply by akarshghale** [reply]
Hi, Thank You for the reply.

I have checked those docs but unfortunately I am still unclear on how can I use openAI embeddings. If you don't mind could you please give a little bit more instructions/examples on how I should go about it?

Thank You! Much appreciated.

**Reply by MaartenGr** [reply, marked as answer]
When you have created your OpenAI embeddings through something like the following:

```python
embeddings = [r["embedding"] for r in openai.Embedding.create(input=docs, model=model)["data"]]
```

You can then pass them as follows:

```python
topics, probs = topic_model.fit_transform(docs, embeddings)
```

The embeddings are a numpy array where each row is an embedding for each document.

**Reply by akarshghale** [reply]
Got it! Thank you. Was just confused about the data structure. It's clear now!
